theurgy.lic

```diff
@@ -2,185 +2,453 @@
-
-  def check_supplies
-    ensure_copper_on_hand(600, @hometown)
-    if DRStats.circle > 60
-      buy_taffelberries unless inside?('Taffelberries', @theurgy_supply_container)
+  
+  def commune_kertigen
+    if inside_container?('holy oil')
+      get_from_container('holy oil')
+    else
+      get_from_container('some oil')
+      complete_or_interrupt_research
+      cast_spell({'abbrev' => 'bless',
+                  'mana' => 1,
+                  'prep_time' => 5,
+                  'cast' => 'cast oil'}, @settings)
+      continue_research
+    end
+    
+    if right_hand.include? 'holy oil'
+      fput "sprinkle oil on #{checkname}"
+      put_in_container('oil')
+      match = bput('commune kertigen',
+                   'completed this commune too recently',
+                   'The thick smell of ozone fills your nostrils')
+      return match =~ /ozone/
+    else
+      stow_hands
+      return false
     end
-    buy_theurgy_supply('incense')
-    buy_theurgy_supply('sage')
-    buy_theurgy_supply('lavender')
-    buy_theurgy_supply('wine')
-    buy_oil unless inside?('holy oil', @theurgy_supply_container) || inside?('some oil', @theurgy_supply_container)
-    buy_flint unless inside?('flint', @theurgy_supply_container)
-    buy_parchment unless inside?('golden parchment', @theurgy_supply_container)
   end
 
-  def buy_theurgy_supply(item_name)
-    return if inside?(item_name, @theurgy_supply_container)
-
-    room = get_data('town')[@hometown]['theurgy_supplies']['id']
-    buy_item(room, item_name)
-    put_in_container(item_name)
+  ### RITUALS ###
+  def perform_next_ritual
+    perform_ritual(@rituals.shift) if rituals_remain?
   end
 
-  def buy_flint
-    buy_item(8265, 'flint')
-    put_in_container('flint')
+  def perform_ritual(ritual)
+    send(ritual[:method])
   end
 
-  def buy_oil
-    buy_item(19_076, 'oil')
-    put_in_container('oil')
-  end
+  def tithe
+    if UserVars.tithe_timer
+      return false unless Time.now - UserVars.tithe_timer > 60 * 10
+    end
+    return false unless withdraw_exact_amount?('5 silver', @hometown)
 
-  def buy_parchment
-    return unless ensure_copper_on_hand(6300, @hometown)
-    walk_to('5735')
-    fput 'read placard'
-    pause 1
-    fput 'order 4 from monk'
-    fput 'unroll my parchment'
-    put_in_container('golden parchment')
+    walk_to @data['almsbox']['id']
+    UserVars.tithe_timer = Time.now
+    # TODO: does almsbox name vary by location?
+    bput("put 5 silver #{hometown_currency(@hometown)} in almsbox",
+         'You drop', 'But you do not')
   end
 
-  def buy_taffelberries
-    walk_to(8280)
-    bput('order 14', 'puts some sugar-dipped taffelberries on the bar', 'You don\'t have enough money to afford that')
-    fput 'get Taffelberries'
-    put_in_container('Taffelberries')
+  def study_wall
+    get_data('spells').spell_data do |name, properties|
+      if properties['cyclic'] && DRSpells.active_spells.keys.include?(name)
+        return false
```

Comment:
Is the intent here to _not_ study the wall if you have _any_ cyclic active?

Reply:
Yes, I believe that's correct.  This is copied from the original script and I can't test this ritual myself (I think it maybe requires having rezz?)